#Analysis 7 - Getting a closer look at the clusters from the orignal clusters. 

##Purpose

To get start to understand the differences in GO categories between the clusters. This is the data from analysis1D. 

###Part 1

This will look into the number of genes that are the same between the clusters and the genotypes. With some basic visualization.

Required Libraries

```{r}
library(VennDiagram)
library(ggplot2)
library(reshape)
library(kohonen)
library(goseq)
library(GO.db)
```

###Visualize by Cluster
Read in data used for GO enrichment analysis

Read in data produced from analysis1D.
```{r}
plot.data <- read.table("../data/analysis1.som.data.small.ALLD.txt",header=TRUE)
names(plot.data)

levels(plot.data$genotype) #double check that I am reading the correct data set.
```


##Cluster Specific analysis
Now I want to take a look at what are is going on exactly in these clusters. The clusters start with the bottom left, which is cluster number 1.

This is a function that makes a boxplot showing the transformed values of expression in the clusters. 

```{r}
#clusterVis Function
#displays transformed data in a box plot and 
clusterVis <- function(clustNum){
  
  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  sub_data <- sub_cluster[,9:14] # just the sample types
  m.data <- melt(sub_data) 
  
  p <- ggplot(m.data, aes(x=variable, y=value))
  p + geom_point(alpha=0.5, position="jitter", size=1) + geom_boxplot(alpha=0.75, outlier.size=0) 
}
```

#clusterVis_geno Function
#displays transformed data in a box plot and genotype information 
```{r}
clusterVis_geno <- function(clustNum){
  
  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  sub_data <- sub_cluster[,c(1,9:14)] # just the sample types
  
  m.data <- melt(sub_data) 
  
  m.data$genotype <- as.factor(m.data$genotype)
  
 # m.data$region <- ifelse(grepl("A", m.data$variable, ignore.case = T), "A.tip", 
                        # ifelse(grepl("B", m.data$variable, ignore.case = T), "B.middle", "C.base"))
 # m.data$tissue <- ifelse(grepl("other", m.data$variable, ignore.case = T), "rachis", 
                          #ifelse(grepl("mbr", m.data$variable, ignore.case = T), "margin", "base"))
  
  p <- ggplot(m.data, aes(x=variable, y=value, color = genotype))
  p + geom_point(alpha=0.5, position="jitter", size=1) + 
    geom_boxplot(alpha=0.75, outlier.size=0) + 
    theme_bw() + 
    scale_colour_manual(values=c("#ef8a62", "#67a9cf")) 

}
```

Number of genes function, which gives you some basics about the clusters.

```{r}
clusterNum <- function(clustNum){

  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  print(paste("total number of genes in sub cluster is ", 
              nrow(sub_cluster)
              )
        )

  scwt <- subset(sub_cluster, genotype == "wt")
  print(paste("total number of genes in wt cluster is ", 
              nrow(scwt)
              )
        )

  sctf2 <- subset(sub_cluster, genotype == "tf2")
  print(paste("total number of genes in tf2 cluster is ", 
              nrow(sctf2)
              )
        )
  
  scIntersect <- as.data.frame(intersect(scwt$gene, sctf2$gene))
   print(paste("There are", 
               length(intersect(scwt$gene, sctf2$gene)), 
               " that are the same between wt and tf2"
               )
         )
  
  ##Venn Diagram part
  grid.newpage()
  venn.plot <- draw.pairwise.venn(area1 = nrow(scwt),
                              area2    = nrow(sctf2),
                              cross.area = length(intersect(scwt$gene, sctf2$gene)),
                              scaled       = F,
                              category     = c("Wildtype", "tf2"),
                              fill         = c("blue", "red"),
                                alpha        = 0.3,
                                lty          = "blank",
                                cex          = 2,
                                cat.cex      = 2,
                                cat.pos      = c(315, 25),
                                cat.dist     = 0.09,
                                cat.just     = list(c(-1, -1), c(1, 1)),
                                ext.pos      = 30,
                                ext.dist     = -0.05,
                                ext.length   = 0.85)
  grid.draw(venn.plot)

}
```


```{r}
clusterGO <- function(clustNum){
##GO Enrichment on the catergories

#we need to first get the data in the right format.
#First get the list of ITAG, 

#sub_cluster
sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
scwt <- subset(sub_cluster, genotype == "wt")
sctf2 <- subset(sub_cluster, genotype == "tf2")
scIntersect <- as.data.frame(intersect(scwt$gene, sctf2$gene))

itag.sc <- as.data.frame(sub_cluster$gene) 
colnames(itag.sc)[1] <- "itag"
itag.sc$sc <- 1  

#scwt
itag.scwt <- as.data.frame(scwt$gene) 
colnames(itag.scwt)[1] <- "itag"
itag.scwt$wt <- 1  

#sctf2
itag.sctf2 <- as.data.frame(sctf2$gene) 
colnames(itag.sctf2)[1] <- "itag"
itag.sctf2$tf2 <- 1  

#Intersect
itag.scIntersect <- as.data.frame(scIntersect[1]) 
colnames(itag.scIntersect)[1] <- "itag"
itag.scIntersect$intersect <- 1 

#Merge all by itag
ITAGmerge <- merge(itag.scIntersect, itag.scwt, by = "itag", all= TRUE)
ITAGmerge <- merge(ITAGmerge, itag.sctf2, by = "itag", all = TRUE)
matrixGO <- merge(ITAGmerge, geneLength, by = "itag", all = TRUE)
matrixGO[is.na(matrixGO)] <- 0
pat <- matrixGO

#Now that we have the data in the right format we can proceed with GO enrichment.

#First specify vector to loop over for each column

sigType <- c("intersect", "wt", "tf2")

  for(type in sigType) {
  
    genes = as.integer(pat[,type])
    names(genes) = pat$itag
    table(genes)
    length(genes)
  
    pwf = nullp(genes,bias.data=pat$length)
  
    GO.wall = goseq(pwf,gene2cat = cate)
    head(GO.wall)
  
  #This is going to correct for multiple testing.  You can specify the p-value cut-off of GO categories you are interested.
  
    enriched.GO = GO.wall$category[p.adjust(GO.wall$over_represented_pvalue, method = "BH") < 0.05]
  
    enriched.GO
  
    my.GO <- as.character(enriched.GO)
    my.GO.table <- Term(my.GO)
    my.GO.table
    t <- as.matrix(my.GO.table)
  
    print(type) #this is for the knitr document
    print(t) #this is for the knitr document
  }
}
```

```{r}
#Prereq annotation files for function


geneLength <- read.csv("../../../07GO_enrichment/requisiteData/normalized_genes_length.csv")
cate <- read.table("../../../07GO_enrichment/requisiteData/melted.GOTable.txt",header=TRUE)

annotation1<- read.delim("../../../06diffGeneExp/analysis/data/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2<- read.delim ("../../../06diffGeneExp/analysis/data/ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")
#Only Gene Name and ITAG
annotation <- annotation[,c(1,3,5)]
#fix to one regex
annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)
annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)

###genesInClust()

#This looks at how many unique genes are in each cluster. 

genesInClust_tf2 <- function(clustNum) {
  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  
  #Get rid of the end numbers
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  
  sctf2 <- subset(sub_cluster, genotype == "tf2")
  sub_data <- as.data.frame(sctf2[,2])
  colnames(sub_data) <- "ITAG"
  
  resultsTable <- merge(sub_data,annotation,by = "ITAG", all.x=TRUE)
  print(nrow(unique(resultsTable)))
  return(unique(resultsTable))
  }

genesInClust_wt <- function(clustNum) {
  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  
  #Get rid of the end numbers
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  
  scwt <- subset(sub_cluster, genotype == "wt")
  sub_data <- as.data.frame(scwt[,2])
  colnames(sub_data) <- "ITAG"
  
  resultsTable <- merge(sub_data,annotation,by = "ITAG", all.x=TRUE)
  print(nrow(unique(resultsTable)))
  return(unique(resultsTable))
  }

genesInClust_intersect <- function(clustNum) {
  sub_cluster <- subset(plot.data, som.unit.classif==clustNum)
  
  #Get rid of the end numbers
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  sub_cluster$gene <- gsub("^(.*)[.].*", "\\1", sub_cluster$gene)
  
  sctf2 <- subset(sub_cluster, genotype == "tf2")
  scwt <- subset(sub_cluster, genotype == "wt")
  scIntersect <- as.data.frame(intersect(scwt$gene, sctf2$gene))
  dim(scIntersect)
  sub_data <- as.data.frame(scIntersect[,1])
  colnames(sub_data) <- "ITAG"
  resultsTable <- merge(sub_data,annotation,by = "ITAG", all.x=TRUE)
  print(nrow(unique(resultsTable)))
  return(unique(resultsTable))
  }
```



###Cluster 1

Sub cluster 1 is defined by up regulation of genes in Bmbr, which is the early leaflet region of the terminal leaflet. Very Simialar to GO terms found from DE expression analsysis. 

```{r}
clusterVis(1)
clusterVis_geno(1)

#Choose which one you are interested in and uncomment 
#print to screen in a readible manner with kable()

#y <- genesInClust_tf2(1)
#y <- genesInClust_wt(1)
#y <- genesInClust_intersect(1)

kable(y, format = "latex", booktabs = TRUE)

clusterNum(1)
clusterGO(1)
```

###Cluster 2

Sub cluster 2 is defined by up regulation of genes in Cmbr, which is the base "marginal blastozone" region, which should be the most pluripotent in WT. Same as DE (wtcmbr vs wtco) except with less categories. 


##Wildtype specific
Solyc09g083280 & AT5G43700 & Auxin inducible protein similar to transcription factors.\\

Solyc08g016070 & AT4G16340 & mutant has Seedling lethal; Cotyledon, leaf-shape, trichome defects; Putative Cytoskeletal Protein\\

Solyc07g062840 & AT5G53950 & Transcriptional activator of the NAC gene family, with CUC1 redundantly required for embryonic apical meristem formation, cotyledon separation and expression of STM.  Proper timing of CUC2 expression is required to maintain the phyllotactic pattern initiated in the meristem.

Solyc12g077670 & AT1G18800 & Double nrp1-1 nrp2-1 mutants show arrest of cell cycle progression at G2/M  and disordered cellular organization occurred in root tips. Localize in the nucleus and can form homomeric and heteromeric protein complexes with NRP1. Bind histones Histone2A and Histone2B and associate with chromatin in vivo.\\

##tf2 specific

Solyc12g006340 & AT1G30330 & Encodes a member of the auxin response factor family. Mediates auxin response via expression of auxin regulated genes. Acts redundantly with ARF8 to control stamen elongation and flower maturation. Expression of ARF6 is controlled by miR167.\\

Solyc11g013150 & AT4G08250 & scarecrow transcription factor family protein; similar to SCR (SCARECROW), transcription factor [Arabidopsis thaliana] (TAIR:AT3G54220.1); similar to Nodulation signaling pathway 2 protein (GB:Q5NE24); contains InterPro domain GRAS transcription factor; (InterPro:IPR005202)\\

Solyc08g079100 & AT4G00180 & YABBY gene family member, likely has transcription factor activity, involved in specifying abaxial cell fate. Along with FIL, involved in patterning of the fruit. GUS reporter gene expression in seedlings is observed in the young leaves and as the leaf matures, expression is restricted to the abaxial tissues of leaves, expression is also observed on either side of the leaf margin in the younger tissues of leaf blades.\\

Solyc07g045550 & AT5G35770 & A recessive mutation in the Arabidopsis STERILE APETALA (SAP) causes severe aberrations in inflorescence  and flower and ovule development.\\

Solyc01g100510 & AT1G23380 & homeodomain transcription factor KNAT6, belonging to class I of KN transcription factor family (which also includes KNAT1 and KNAT2). Expression is increased in as and bop1 leaf mutants.\\




```{r}
clusterVis(2)
clusterVis_geno(2)

w <- genesInClust_tf2(2)
x <- genesInClust_wt(2)
y <- genesInClust_intersect(2)

z <- setdiff(w$ITAG,y$ITAG)
z <- as.data.frame(z)

names(z)[1]<- "ITAG"
head(z)
resultsTable <- merge(z,annotation,by = "ITAG", all.x=TRUE)

kable(resultsTable, format = "latex", booktabs = TRUE)
dim(resultsTable)
kable(x, format = "latex", booktabs = TRUE)

clusterNum(2)
clusterGO(2)
```

###Cluster 3

This cluster is specific to Cother, which is specific to the rachis region at the base. 

```{r}
clusterVis(3)
clusterVis_geno(3)
clusterNum(3)
clusterGO(3)
```

###Cluster 4 

This cluster has genes that are preferentially up-regulated in Ambr, which is the tip most region that becomes the terminal leaflet.  This is the terminal leaflet blade region 
```{r}
clusterVis(4)
clusterVis_geno(4)
clusterNum(4)
clusterGO(4)
```

###Cluster 5 

The cluster ihas genes that are preferentially up-regulated in Aother, which is the rachis region at the tip; what will eventually become the midvien of the terminal leaflet.

```{r}
clusterVis(5)
clusterVis_geno(5)

y <- genesInClust_tf2(5)
y <- genesInClust_wt(5)
y <- genesInClust_intersect(5)

kable(y, format = "latex", booktabs = TRUE)

clusterNum(5)
clusterGO(5)
```

###Cluster  6

The cluster ihas genes that are preferentially up-regulated in Bother, which is the rachis region at site of leaflet initiation.

```{r}
clusterVis(6)
clusterNum(6)
clusterGO(6)
```
