#Analysis 9 - Getting a closer look at the clusters from the clusters from basic some WT only - Small

##Purpose 

This analysis is based off of `lcmSOM_analysis1.0a_071014.Rmd`, where the original dataset was made.  The most DE genes of the WT genotype were subsetted, where mean was taken from the reps.  The purpose is to find trends in the wildtype only data.

##Questions

1. What type of gene co-expression patterns are found in this data?
2. Is there any cluster that is enriched in genes from the curated leaf gene list?
3. Are there GO categories that are enriched in the clusters? 
4. If there are GO categories, what are the genes that are represented?

##Caveats

1. While the standard deviation of expression of each gene in each tissue were from the reps were calculated, this was not used when creating the self organized map. 


##Analysis Set-up

Required Libraries

```{r}
library(VennDiagram)
library(ggplot2)
library(reshape)
library(kohonen)
library(goseq)
library(GO.db)
source("./clusterFunctions.R")
```

Upload that dataset:

```{r}
gene25 <- read.csv("../data/allGeneList.csv")


#Principle Component Analysis
pca <- prcomp(scale_data, scale=TRUE) 
summary(pca) 
pca.scores <- data.frame(pca$x)

#Bring the data back together
data.val <- cbind(mostDEgene.long, scale_data, pca.scores) 
```



##Self Organizing Map - (3,2) Small

```{r}
#subset only the scaled gene expression values
som.data <- as.matrix(data.val[,c(8:13)])  

#Check that you are using the tranformed values of each tissue type

head(som.data)


set.seed(6)
som <- som(data=som.data, somgrid(3,2,"hexagonal")) # This is where you change the size of the map
summary(som)
```


###Look at the SOM results
```{r}
plot(som, type = "codes")
plot(som, type = "counts")
plot(som, type="dist.neighbours")

data.val2 <- cbind(data.val,som$unit.classif,som$distances) #bring 

#fix to one regex
data.val2$gene <- gsub("^(.*)[.].*", "\\1", data.val2$gene)
data.val2$gene <- gsub("^(.*)[.].*", "\\1", data.val2$gene)

names(data.val2)[20] <- "unit.classif"
```

Visualize the major clusters. 
```{r}
data.val2$unit.classif <- as.factor(data.val2$unit.classif)
p <- ggplot(data.val2, aes(PC1, PC2, color = unit.classif)) 

p + geom_point(size=I(3), alpha = 0.6) +
  theme_bw()
```

###Visualize by Cluster
Read in data used for GO enrichment analysis.

```{r}
geneLength <- read.csv("../../../07GO_enrichment/requisiteData/normalized_genes_length.csv")
cate <- read.table("../../../07GO_enrichment/requisiteData/melted.GOTable.txt",header=TRUE)
```

###Cluster 1
base rachis specific
```{r}
clusterVis(1)
clusterVis_PCA(1)
clusterGO(1)
y <- genesInClust(1, data.val2, annotation)

#If value intersects with an
intersect(y$ITAG, genesOfInterest$genes)
```

###Cluster 2

```{r}
plot(som, type = "codes")
clusterVis_PCA(2)
clusterVis(2)
clusterGO(2)
y <- genesInClust(2, data.val2, annotation)
````
###Cluster 3 

```{r}
clusterVis_PCA(3)
clusterVis(3)
clusterGO(3)
```

###Cluster 4

```{r}
clusterVis_PCA(4)
clusterVis(4)
clusterGO(4)
y <- genesInClust(4, data.val2, annotation)
```

###Cluster 5  
Upregulation of BOther, whith many GO categories


```{r}
clusterVis_PCA(5)
clusterVis(5)
clusterGO(5) 
y <- genesInClust(5, data.val2, annotation)
```

###Cluster  6

```{r}
clusterVis_PCA(6)
clusterVis(6)
clusterGO(6)
```
