#Getting the most sig. differentiated genes between tissue

Now I am doing exactly what I did in Analysis1 but with *tf2*. I will do one with *tf2* specific genes and "everything" which is from the analysis5.  


##Required Libraries

```{r}
library(ggplot2)
library(reshape)
library(plyr)
library(kohonen)
```

Read in files and format data from raw files
=============================================

Create a list of all DE analysis:

###Count Data
```{r, eval=FALSE}
#Read in count data
countData <- read.csv("../data/normalized_read_count.csv")
#Melt count data
countData <- melt(countData)
colnames(countData) <- c("gene", "sample", "count")

#set genotype

countData$genotype <- ifelse(grepl("wt", countData$sample, ignore.case = T), "wt", 
         ifelse(grepl("tf2", countData$sample, ignore.case = T), "tf2", "unknown"))

#set tissue

countData$tissue <- ifelse(grepl("other", countData$sample, ignore.case = T), "other", 
         ifelse(grepl("mbr", countData$sample, ignore.case = T), "mbr", "unknown"))

#Set Region
countData$region <- ifelse(grepl("a", countData$sample, ignore.case = T), "A", 
         ifelse(grepl("c", countData$sample, ignore.case = T), "C", "B"))

#Set type

countData$type <- paste(countData$region, countData$tissue,  sep = "")
head(countData)

#Subset by Genotype, since we will be looking at only tf2 at this stage
countData <- subset(countData, genotype == "tf2")
```

###Most Significantly DE genes in *tf2*
```{r, eval = FALSE}
#Read in each list of DE expressed genes
tf2aother_tf2cother  <- read.table("../data/allSigGenes/tf2aother_tf2cother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2ambr_tf2aother <- read.csv("../data/allSigGenes/tf2ambr_tf2aother_DE_sig.csv")
tf2ambr_tf2bmbr <- read.table("../data/allSigGenes/tf2ambr_tf2bmbr_DE_sig.txt", header = TRUE, fill = TRUE)
tf2ambr_tf2cmbr <- read.table("../data/allSigGenes/tf2ambr_tf2cmbr_DE_sig.txt", header = TRUE, fill = TRUE)
tf2aother_tf2bother <- read.table("../data/allSigGenes/tf2aother_tf2bother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2bmbr_tf2bother <- read.table("../data/allSigGenes/tf2bmbr_tf2bother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2bmbr_tf2cmbr <- read.table("../data/allSigGenes/tf2bmbr_tf2cmbr_DE_sig.txt", header = TRUE, fill = TRUE)
tf2bother_tf2cother <- read.table("../data/allSigGenes/tf2bother_tf2cother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2cmbr_tf2cother <- read.table("../data/allSigGenes/tf2cmbr_tf2cother_DE_sig.txt", header = TRUE, fill = TRUE)

#merge them together
allGenes <- rbind(tf2aother_tf2cother,tf2ambr_tf2aother$ITAG,tf2ambr_tf2cmbr, tf2aother_tf2bother,tf2bmbr_tf2cmbr,tf2bother_tf2cother,tf2cmbr_tf2cother)

#tf2ambr_tf2aother_DE_sig.txt!  What the hell is wrong with this file? tf2ambr_tf2bmbr$ITAG?

The work around is to scrape the data seperatly 

dim(allGenes)
head(allGenes)

allGenes$ITAG

#recieve just the list
allGenesITAG <- allGenes[,1]
length(allGenesITAG)
#Remove duplicates
allGenesITAG <- unique(allGenesITAG)
head(allGenesITAG)

#make an empty table to hold all the genes
allGeneList <- data.frame(t(rep(NA,7)))
colnames(allGeneList) <- c("type", "genotype", "N", "mean", "sd", "se", "gene")
allGeneList <- allGeneList[-1,] #remove first row

```

#Loop together all relevent gene information.

```{r, eval = FALSE}
for(GENE in allGenesITAG) {
 
  if(length(grep(GENE, countData$gene)) < 1){ #this is just making sure that the list of sig genes
    next; 
    }
      
  geneData <- subset(countData, grepl(GENE, countData$gene))
   
  sumGraph <- ddply(geneData, c("type", "genotype"), summarise,
               N    = length(count),
               mean = mean(count),
               sd   = sd(count),
               se   = sd / sqrt(N))
  
  sumGraph$gene <- GENE
  
allGeneList  <- rbind(allGeneList, sumGraph) #bind together all the new rows per loop. 
}
dim(allGeneList)
head(allGeneList)
write.table(allGeneList, file = "../data/allGeneList.tf2only_analysis1.5.csv", sep = ",")
```


Self Organizing Maps 
========================================================

The goal of this analysis is to find genes that have co-expression patterns of 

1. most differentiated genes between tissue. 

##1.pca.R

First read in file that came from mostSigDEgenes.Rmd.  This is a list of genes from all DE analysis in WT. They were all cancatenated, then duplicate genes were removed.  In addition the mean was calculated from the replicates of each type. 

The first step is to get it into the right format. First column being the genes, while the subsequent columns are the different libraries (type).

```{r}
mostDEgenes <- read.csv("../data/allGeneList.tf2only_analysis1.5.csv")
head(mostDEgenes)

mostDEgenes <- mostDEgenes[c(7, 1, 4)] #keep only needed columns (gene, type, mean)

#Change from long to wide data format
mostDEgene.long <- cast(mostDEgenes, gene ~ type, value.var = mean, fun.aggregate = "mean")  #why did I have to specify "mean" here? Are there duplicates of types? Double check later. 
mostDEgene.long <- as.data.frame(mostDEgene.long). 
scale_data <- as.matrix(t(scale(t(mostDEgene.long[c(2:7)])))) #transformation

#Principle Component Analysis
pca <- prcomp(scale_data, scale=TRUE) 

summary(pca) 

pca.scores <- data.frame(pca$x)

data.val <- cbind(mostDEgene.long, scale_data, pca.scores) 
head(data.val)
```


##Visualizing the PCA

Looks to be three major clusters. 

```{r}
p <- ggplot(data.val, aes(PC1, PC2)) 
p + geom_point()
```

#1. Self Organizing Map - (6,6) Large

The size of the map is something that may cause differences in the genes that are clustered.  Using a small map size (3,2), I found they cluster in according to tissue type. This makes the interpretation of the results pretty straight forward. My only worry is that the map might not be large enough, considering [1], suggests that you size of the map based on count distribution, the goal being an even distribution, with no "peak" counts in any one cluster while also having no empty clusters.  

The only way to see how this is affects what we see is to compare the clusters of the small (3,2) and large map (6,6).  

```{r}
names(data.val)

som.data <- as.matrix(data.val[,c(8:13)])  #subset only the scaled gene expression values

set.seed(4)

som <- som(data=som.data, somgrid(6,6,"hexagonal")) # This is where you change the size of the map
summary(som)
```

###Training Plot ("changes") - Large

This shows a hundred iterations. Training decreases with iterations and plateaus at around 40 iterations.  Ideally you want the the training to reach a minimum plateau.  In the example online, the decrease to this plateau happens slowly with a slow decline to the minimum plateau. 

```{r}
plot(som, type ="changes")
```

###Code Plot - Large

The the code plot shows each cluster and the node wieght vectors or "codes" associated with each node. These are made up of the original normalized values of the original values used to generate the map.  You should see patterns of clustering.

The fan chart in the center of the clusters reveals the characteristics that define how the genes were clustered into each particular cluster.  For instance if one cluster has only one large fan piece, say for Bother, this is telling us that most of the genes in this cluster were grouped because of similar normalized gene count value of the Bother region. We do not know the degree, it could mean all these genes are up-regulated or down-regulated in the Bother region, but we do not know which at this point. 

```{r}
plot(som, type = "codes")
```

###Count Plot - Large

This tells you how many genes are in each of the clusters. The count plot can be used as a quality check.  Ideally you want a uniform distribution.  If there are some peaks in certain areas, this means you should likely increase the map size.  If you have empty nodes you should decrease the map size [1]. 

```{r}
plot(som, type = "counts")
```

###Distance Neighbour Plot - Large

This is sometimes called the "U-Matrix", it can help identify further clustering. Areas of low neighbour distance indicate groups of nodes that are similar and the further apart nodes indicate natural "borders" in the map. 

```{r}
plot(som, type="dist.neighbours")
```

###Heatmaps - large

This shows the distribution of each type 

```{r}
head(som$codes)
som$data <- data.frame(som$data) #changed to dataframe to extract column names easier. 

#This is just a loop that plots the distribution of each tissue type across the map. 
for (i in 1:6){
  plot(som, type = "property", property = som$codes[,i], main=names(som$data)[i])
  print(plot)
  }
```

###Clustering Plot - Large

This groups clusters based on similar "metrics".  The advice given from [1], suggests that the heatmap should be used to view the overall "story" of the map. Essentiatlly you are taking all the tissue types into account and viewing it all on one heat map. 

```{r}
#"estimate of the number of clusters that would be suitable can be ascertained using a kmeans algorithm and examing for an “elbow-point” in the plot of “within cluster sum of squares[1]”

#What the hell is going on here?

mydata <- som$codes
head(mydata)

wss <- (nrow(mydata)-1)*sum(apply(mydata,2,var)) 

for (i in 1:8) {
  wss[i] <- sum(kmeans(mydata, centers=i)$withinss)
}

plot(wss)
```

This is setting the larger clusters that incorporate multiple clusters. 

```{r}

## use hierarchical clustering to cluster the codebook vectors
som_cluster <- cutree(hclust(dist(som$codes)), 6)
# plot these results:
plot(som, type="mapping", bgcol = som_cluster, main = "Clusters") 
add.cluster.boundaries(som, som_cluster) 

# I want to attach the hierchal cluster to the larger dataset data.val.
som_clusterKey <- data.frame(som_cluster)
som_clusterKey$unit.classif <- c(1:36)

data.val <- cbind(data.val,som$unit.classif,som$distances)

#Merge data.val with som_clusterKey
##change data.val to match som_cluster key 
names(data.val)[20] <- "unit.classif"

data.val <- merge(data.val, som_clusterKey, by.x = "unit.classif" ) #ignore warning, this is what you want.  You are essentially filling in the empties with the value from som_clusterKey

write.table(data.val, file="../data/analysis1.5.som.data.txt")
#Make sure that there is just one of each value som$unit.classif and distances column. 
names(data.val)
```

#Other Visualization - Large

###Visualize by Cluster

```{r}
plot.data <- read.table("../data/analysis1.5.som.data.txt",header=TRUE)
names(plot.data)
dim(plot.data)

p <- ggplot(plot.data, aes(PC1, PC2, colour=factor(som_cluster))) #notice I am using som_cluster and not unit.classif, because this is the larger SOM
p + geom_point() + theme_bw()
```

###Visualize by individual clusters - Large

What the hell is going on here for the large cluster?

```{r}
sub_cluster <- subset(plot.data, som_cluster =="5") #Again here I am interested in the clusters made up of many cluster and am not using unit.classif

sub_data <- sub_cluster[,8:13] # just the sample types

names(sub_data)
head(sub_data)
m.data <- melt(sub_data) 

head(m.data)

p <- ggplot(m.data, aes(x=variable, y=value))
p + geom_point(alpha=0.5, position="jitter", size=1) + geom_boxplot(alpha=0.75, outlier.size=0) 
```

Conclusion:

It appears as though we get clustering in tissue specific regions. When you set the size of the map to six, you get six clusters that are tissue specific regions.  The problem comes when that there is a "peak" in the counts data and ideally you want a more even distribution according to the tutorial [1].  I then made the map bigger and was able down the line to get 5 distinct clusters.  The question is are these the same clusters that were in the original 3,3 map?  I will have to isolate the each of the clusters in both circumstances and see if I have the same genes. It could be that using more clusters will help identify more interaction between the tissue.  For instance, genes in one larger cluster migh be comprised of a group of genes that is upregulated 



2. Self Organizing Map- Small (3,2)
-----------------------------

The size of the map is something that may cause differences in the genes that are clustered.  Using a small map size (3,2), I found they cluster in according to tissue type. See below.

```{r}

som.data <- as.matrix(data.val[,c(8:13)]) 

som.data <- as.matrix(data.val.small[,c(8:13)])  #subset only the scaled gene expression values

set.seed(2)

som <- som(data=som.data, somgrid(3,2,"hexagonal")) # This is where you change the size of the map
summary(som)
```

###Training Plot ("changes") - Small

This shows a hundred iterations. Training plateaus at around 20 iterations steeply.  Is this a problem?

```{r}
plot(som, type ="changes")
```

###Code Plot - Small

Here with the small map, each tissue has a tissue specific cluster. 

```{r}
plot(som, type = "codes")
```

###Count Plot - Small

This tells you how many genes are in each of the clusters. The Aother cluster has around 600 genes, which may be a little high. 

```{r}
plot(som, type = "counts")
```

###Distance Neighbour Plot- Small

This is sometimes called the "U-Matrix", it can help identify further clustering. Areas of low neighbour distance indicate groups of nodes that are similar and the further apart nodes indicate natural "borders" in the map. 

```{r}
plot(som, type="dist.neighbours")
```

###Heatmaps - Small 

This shows the distribution of each type of tissue.  This doesn't really work too well when the the map is so small. Bother is the only tissue type that contributes to two clusters. 

```{r}
head(som$codes)
som$data <- data.frame(som$data) #changed to dataframe to extract column names easier. 

#This is just a loop that plots the distrinution of each tissue type across the map. 
for (i in 1:6){
  plot(som, type = "property", property = som$codes[,i], main=names(som$data)[i])
  print(plot)
  }
```

Output
```{r}
data.val.small <- cbind(data.val.small,som$unit.classif,som$distances)
write.table(data.val.small, file="../data/analysis1.5.som.data.small.txt")
#Make sure that there is just one of each value som$unit.classif and distances column. 
names(data.val.small)
```

#Other Visualization 

###Visualize by Cluster

```{r}
plot.data <- read.table("../data/analysis1.5.som.data.small.txt",header=TRUE)
names(plot.data)

p <- ggplot(plot.data, aes(PC1, PC2, colour=factor(som.unit.classif))) #use unit.classif for smaller dataset
p + geom_point() + theme_bw()
```

###Visualize by individual clusters


```{r}
sub_cluster <- subset(plot.data, som.unit.classif=="6")

sub_data <- sub_cluster[,8:13] # just the sample types

names(sub_data)
head(sub_data)
m.data <- melt(sub_data) 

head(m.data)

p <- ggplot(m.data, aes(x=variable, y=value))
p + geom_point(alpha=0.5, position="jitter", size=1) + geom_boxplot(alpha=0.75, outlier.size=0) 
```

Conclusion:

Just looking at the PCA between the two, they show the same clusters no matter if you do the small or large.  So in light of this, I am going to go ahead and continue with the small because it is more intuitive to understand and essentially they are the same thing. 

##References

1. [R self Organizing map tutorial](http://www.r-bloggers.com/self-organising-maps-for-customer-segmentation-using-r/)
