#Analysis 9 - Getting a closer look at the clusters from the clusters from basic some WT only.  

##Create the SOM 

This analysis is based off of `lcmSOM_analysis1.0a_071014.Rmd`, where the original dataset was made.  The most DE genes in the WT category were subsetted, where mean was taken from the reps.  ONLY WILDTYPE HERE.

Upload that dataset:

```{r}
mostDEgenes <- read.csv("../data/allGeneList.csv")

mostDEgenes <- mostDEgenes[c(7, 1, 4)] #keep only needed columns (gene, type, mean)

#Change from long to wide data format
mostDEgene.long <- cast(mostDEgenes, gene ~ type, value.var = mean, fun.aggregate = "mean")  #why did I have to specify "mean" here? Are there duplicates of types? Double check later. 
mostDEgene.long <- as.data.frame(mostDEgene.long)  
scale_data <- as.matrix(t(scale(t(mostDEgene.long[c(2:7)]))))#transformation.

#Principle Component Analysis
pca <- prcomp(scale_data, scale=TRUE) 

summary(pca) 

pca.scores <- data.frame(pca$x)

data.val <- cbind(mostDEgene.long, scale_data, pca.scores) 
head(data.val)

```


Visualize the major clusters. 
```{r}
p <- ggplot(data.val, aes(PC1, PC2)) 
p + geom_point()
```

##1. Self Organizing Map - (6,6) Large

```{r}
names(data.val)
dim(data.val)
som.data <- as.matrix(data.val[,c(8:13)])  #subset only the scaled gene expression values
head(som.data)

set.seed(4)

som <- som(data=som.data, somgrid(6,6,"hexagonal")) # This is where you change the size of the map
summary(som)
```

###Look at the SOM results
```{r}
plot(som, type = "codes")
plot(som, type = "counts")
plot(som, type="dist.neighbours")
head(som$codes)
som$data <- data.frame(som$data) #changed to dataframe to extract column names easier. 

data.val <- cbind(data.val,som$som$unit.classif,som$distances) ##########

head(data.val)
data.val$gene <- gsub("^(.*)[.].*", "\\1", data.val$gene)
data.val$gene <- gsub("^(.*)[.].*", "\\1", data.val$gene)

dim(data.val)
#why didn't this work?
#for(GENE in genesOfInterest){
 #data.val$curated <- grepl(GENE, data.val$gene, ignore.case = T)
#}

#This is simple the most stupid way around assigning if a gene is a curated gene!
data.val$curated <- match(data.val$gene, genesOfInterest)
data.val$curated <- gsub("[[:digit:]]+", "yes", data.val$curated)
data.val$curated[is.na(data.val$curated)] <- "no"

#match(data.val$gene, genesOfInterest)
#intersect(data.val$gene, genesOfInterest)
```

##Required Libraries

```{r}
library(VennDiagram)
library(ggplot2)
library(reshape)
library(kohonen)
library(goseq)
library(GO.db)
```


###Visualize by Cluster
Read in data used for GO enrichment analysis

```{r}
geneLength <- read.csv("../../../07GO_enrichment/requisiteData/normalized_genes_length.csv")
cate <- read.table("../../../07GO_enrichment/requisiteData/melted.GOTable.txt",header=TRUE)
```

Read in data produced from analysis1D.
```{r}
plot.data <- data.val
colnames(plot.data)
names(plot.data)
```

##Read in curated gene list

How many genes in the curated gene list are in the most DE expressed genes?
```{r}
geneList1 <- read.csv("../../../06diffGeneExp/analysis/indvGenes/yasuCuratedGenes/pnas.1402835111.sd06.csv")
geneList2 <- read.csv("../../../06diffGeneExp/analysis/indvGenes/yasuCuratedGenes/pnas.1402835111.sd07.csv")

genes2 <- geneList1[,1]
genes1 <- geneList2[,1]

genesOfInterest1 <- levels(genes1)
genesOfInterest2 <- levels(genes2)

genesOfInterest <- c(genesOfInterest1, genesOfInterest2)
genesOfInterest <- genesOfInterest[-1]

genesOfInterest <- as.data.frame(genesOfInterest)
colnames(genesOfInterest) <- c("genes")
head(genesOfInterest)

```
#I need to make a new column in data.val.  If any data.val$gene in genesOfInterest matches data.val$gene place a 1.

##Cluster Specific analysis
Now I want to take a look at what are is going on exactly in these clusters. The clusters start with the bottom left, which is cluster number 1.

This is a function that makes a boxplot showing the transformed values of expression in the clusters. 

```{r}
#clusterVis Function
#displays transformed data in a box plot 
clusterVis <- function(clustNum){
  
  sub_cluster <- subset(plot.data, som$unit.classif==clustNum)
  sub_data <- sub_cluster[,c(1, 8:13)] # just the sample types
  m.data <- melt(sub_data) 
  p <- ggplot(m.data, aes(x=variable, y=value))
  p + geom_point(alpha=0.5,position="jitter", size=1) + 
    geom_boxplot(alpha=0.75, outlier.size=0) + 
    theme_bw()
}

```



```{r}

clusterGO <- function(clustNum){
  ##GO Enrichment on the catergories
   dev.off()
  plot.new()
  
  #we need to first get the data in the right format.
  #First get the list of ITAG
 
  #sub_cluster
  sub_cluster <- subset(plot.data, som$unit.classif==clustNum)
  
  itag.sc <- as.data.frame(sub_cluster$gene) 
  colnames(itag.sc)[1] <- "itag"
  itag.sc$sc <- 1    
 
  #Since each orthologue between tf2 and wt are represented twice in this set, we have to keep only the unique ITAGs.
  
  itag.sc <- unique(itag.sc) #Check. Should cut the list in half. # dim(itag.sc) before and after
  
  #Merge all by itag
  matrixGO <- merge(itag.sc, geneLength, by = "itag", all = TRUE)
  matrixGO[is.na(matrixGO)] <- 0
  pat <- matrixGO
  
  #Now that we have the data in the right format we can proceed with GO enrichment.
  
    genes = as.integer(pat[,"sc"])
    names(genes) = pat$itag
    table(genes)
    length(genes)
  
    pwf = nullp(genes,bias.data=pat$length)
  
    GO.wall = goseq(pwf,gene2cat = cate)
    head(GO.wall)
  
  #This is going to correct for multiple testing.  You can specify the p-value cut-off of GO categories you are interested.
  
    enriched.GO = GO.wall$category[p.adjust(GO.wall$over_represented_pvalue, method = "BH") < 0.05]
  
    enriched.GO
  
    my.GO <- as.character(enriched.GO)
    my.GO.table <- Term(my.GO)
    my.GO.table
    t <- as.matrix(my.GO.table)

    print(t) #this is for the knitr document
}
```


```{r}
clusterVis_line <- function(clustNum) {
  sub_cluster <- subset(plot.data, som$unit.classif==clustNum)
  names(sub_cluster)
  sub_data <- sub_cluster[,c(1,8:13,20)] # just the sample types
  sub_data <- melt(sub_data)
  head(sub_data)
  sub_data <- within(sub_data, lineGroup <- paste(gene,sep='.'))
  ggplot(sub_data, aes(variable, value, group = lineGroup, color = curated)) + 
    geom_line(alpha = .1) + 
    geom_point(alpha = .0) + facet_grid(.~curated)
  }
```

```{r}
#Prereq annotation files for function

annotation1<- read.delim("../../../06diffGeneExp/analysis/data/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2<- read.delim ("../../../06diffGeneExp/analysis/data/ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")
names(annotation)
#Only Gene Name and ITAG
annotation <- annotation[,c(1,5)]

```

Function
```{r}
genesInClust <- function(clustNum, plot.data, annotation) {
  sub_cluster <- subset(plot.data, som$unit.classif==clustNum)
  sub_data <- as.data.frame(sub_cluster[,1])
  colnames(sub_data) <- "ITAG"
  resultsTable <- merge(sub_data,annotation,by = "ITAG", all.x=TRUE)
  print(nrow(unique(resultsTable)))
  return(unique(resultsTable))
  }


```

###Cluster 1
BO specific
```{r}
clusterVis(1)
clusterVis_line(1)
clusterGO(1)
y <- genesInClust(1, plot.data, annotation)

#If value intersects with an
intersect(y$ITAG, genesOfInterest$genes)
```

###Cluster 2

```{r}
clusterVis(2)
clusterVis_line(2)

clusterGO(2)
y <- genesInClust(2, plot.data, annotation)

###Cluster 3 

Ambr in WT is higher than in *tf2* and the WT genes in this cluster have a tight pattern.

```{r}
clusterVis(3)
clusterVis_line(3)
clusterGO(3)
```

###Cluster 4 - Photosynthetic Genes. 

```{r}
clusterVis(4)
clusterVis_line(4)
clusterGO(4)
y <- genesInClust(4, plot.data, annotation)
```

###Cluster 5  
Upregulation of BOther, whith many GO categories


```{r}
clusterVis(5)
clusterVis_line(5)
clusterGO(5) 
y <- genesInClust(5, plot.data, annotation)
```

###Cluster  6



```{r}
clusterVis(6)
clusterVis_line(6)
clusterGO(6)
```

###Cluster  7


```{r}
clusterVis(7)
clusterVis_line(7)
clusterGO(7)
```


###Cluster  8



```{r}
clusterVis(8)
clusterVis_line(8)
clusterGO(8)
```

###Cluster  9


```{r}
clusterVis(9)
clusterVis_line(9)
clusterGO(9)
```

###Cluster  10



```{r}
clusterVis(10)
clusterVis_line(10)
clusterGO(10)
y <- genesInClust(10, plot.data, annotation)
```

###Cluster  11



```{r}
clusterVis(11)
clusterVis_line(11)
clusterGO(11)
y <- genesInClust(11, plot.data, annotation)
```

```{r}
clusterVis(12)
clusterVis_line(12)
clusterGO(12)
y <- genesInClust(12, plot.data, annotation)
```

```{r}
clusterVis(13)
clusterVis_line(13)
clusterGO(13)
y <- genesInClust(13, plot.data, annotation)
```


```{r}
clusterVis(14)
clusterVis_line(14)
clusterGO(14)
y <- genesInClust(14, plot.data, annotation)
```
I need to write something that looks at the statistical difference in tissue between each genotype.  In order to do that I have to subset based on tissue and do a t-test?  Do I need to correct for multiple testing? Ideally what information do I want from these clusters?

1. I want clusters that are enriched in leaf genes. Maybe I could just color special for leaf genes?

2. Significant differences between genotype at each tissue. 







